docker-testing/compose.yaml

```diff
@@ -0,0 +1,88 @@
+# It would be nicer to consolidate the common user/volumes
+# boilerplate, but there wasn't enough time to figure out if it was
+# possible at time of writing. Improvements welcome.
```

Comment:
According to the interwebs, YAML anchors should Just Work

https://nickjanetakis.com/blog/docker-tip-82-using-yaml-anchors-and-x-properties-in-docker-compose

YAML anchors are a plain, standard, built-in YAML feature - nothing specific to Docker at all.

Reply:
According to the same interwebs, you can expand anchors with either `yq explose` or `yaml merge-expand` and make sure you got your YAML anchors right. Either will be orders of magnitude faster than re-creating the image!

Reply:
Also, `yamllint -f parsable file.yaml` has saved me a LOT of YAML time in the past.

(for automation try something like `yamllint -d "{extends: relaxed, rules: {line-length: {max: 100}}}" file.yaml`)